src/github.com/stellar/horizon/ingest/session.go

```diff
@@ -425,7 +418,53 @@ func (is *Session) ingestSignerEffects(effects *EffectIngestion, op xdr.SetOptio
 
 }
 
-func (is *Session) ingestTrades(effects *EffectIngestion, buyer xdr.AccountId, claims []xdr.ClaimOfferAtom) {
+func (is *Session) ingestTrades() {
+	if is.Err != nil {
+		return
+	}
+
+	buyer := is.Cursor.OperationSourceAccount()
+	trades := []xdr.ClaimOfferAtom{}
+
+	switch is.Cursor.OperationType() {
+	case xdr.OperationTypePathPayment:
+		trades = is.Cursor.OperationResult().
+			MustPathPaymentResult().
+			MustSuccess().
+			Offers
+
+	case xdr.OperationTypeManageOffer:
+		trades = is.Cursor.OperationResult().MustManageOfferResult().MustSuccess().OffersClaimed
+	case xdr.OperationTypeCreatePassiveOffer:
+		result := is.Cursor.OperationResult()
+
+		// KNOWN ISSUE:  stellar-core creates results for CreatePassiveOffer operations
+		// with the wrong result arm set.
```

Comment:
It has since been fixed, but unfortunately there are ledgers in the history where this has happened.  I believe nico fixed it back when I discovered the issue.